### PR TITLE
DAOS-9884 Tests: Need to generate pool str from uuid_t in pool_autotest_hdlr()

### PR DESCRIPTION
**Pull request by wiliamhuang**
ap->pool_str was a NULL string which leads to error (container/autotest.py:ContainerAutotestTest.test_container_autotest - DER_INVAL(-1003): Invalid parameters). 

The implementation of dfs_free_sb_layout() was missing which leads to errors of undefine symbol dfs_free_sb_layout.

### CODE REVIEW

File: src/client/dfs/dfs.c [hunk 1]
```diff
@@ -1255,6 +1255,12 @@ dfs_get_sb_layout(daos_key_t *dkey, daos_iod_t *iods[], int *akey_count,
 	return 0;
 }
 
+void
+dfs_free_sb_layout(daos_iod_t *iods[])
+{
+	D_FREE(*iods);
+}
+
```
Comment (daltonbohning) [inline src/client/dfs/dfs.c:1263]: https://github.com/daos-stack/daos/pull/8207 re-adds this, but on a different line. It should be removed here

**Review by daltonbohning — CHANGES_REQUESTED** [review]

**Comment by wiliamhuang** [comment]
All CI checks passed. ContainerAutotestTest ran and passed. Will remove dfs_free_sb_layout() since a recent landed PR contains this function.

File: src/client/dfs/dfs.c (hunk 1 above)
Comment (wiliamhuang) [reply src/client/dfs/dfs.c]: Just revert my change in this file.

**Comment by wiliamhuang** [comment]
@daltonbohning @wangshilong Can you approve this PR? All tests passed yesterday. After that, I merged with master (including submodule update) and removed my implementation of dfs_free_sb_layout in src/client/dfs/dfs.c. Thank you!

**Review by wangshilong — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Comment by wiliamhuang** [comment]
@daos-stack/daos-gatekeeper Can you land this PR? All CI tests passed yesterday. After that, I merged with master (including submodule update) and removed my implementation of dfs_free_sb_layout in src/client/dfs/dfs.c which was added in another PR landed two days ago. Thank you!

**Comment by wiliamhuang** [comment]
@daos-stack/daos-gatekeeper Could you please land this PR ASAP? It is only one line change. I do not think it will break anything.  All CI tests passed two days ago. All CI tests except rebuild_ec in functional hardware medium passed yesterday.  rebuild_ec is a known issue in DAOS-9860. Thank you!

**Comment by wiliamhuang** [comment]
> > @daos-stack/daos-gatekeeper Could you please land this PR ASAP? It is pretty simple and I do not think it will break anything. All CI tests passed two days ago. All CI tests except rebuild_ec in functional hardware medium passed yesterday. rebuild_ec is a known issue in DAOS-9860. Thank you!
> 
> @wiliamhuang for this PR to land it needs to run the test it is fixing. Normally the `Test-tag: pr containerautotest` pragma would be sufficient to run this test but due to https://daosio.atlassian.net/browse/CORCI-1100 and the fact that the `container/autotest.py` uses python 3 code the `list_tests.py` program does not detect the test and it is not run in the intended HW Medium stage. To work around this issue you can add a tag for a test that `list_tests.py` will detect but is currently skipped in the HW Medium stage. In addition I would recommend using the following pragmas to speed up getting a result:
> 
> ```
> Skip-unit-tests: true
> Test-tag: containerautotest online_parallel
> ```

Just started a rebuild with the test-tags you suggested. thx!

**Pull request merged**